server/src/main/scala/za/co/absa/atum/server/api/repository/BaseRepository.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.atum.server.api.repository
+
+import com.github.dwickern.macros.NameOf.nameOf
+import za.co.absa.atum.server.api.exception.DatabaseError
+import za.co.absa.fadb.exceptions.StatusException
+import zio._
+
+trait BaseRepository {
+
+  def handleDbFunctionCall[T, R](
```

Comment:
Maybe `dbCallWithStatus` instead? At this point we have only db functions with statuses, but it won't stay like that forever. So we will have a method for each "database call type".

Reply:
Okay, good prediction / observation :) thanks